### PR TITLE
Return row count: 2 instead of negative row count

### LINKED ISSUE #36143: [MLv2] [BE] [Bug] `underlying-records` drill doesn't handle negative aggregation values in `displayInfo`
This part was missed when producing `displayInfo` for the drill https://github.com/metabase/metabase/blob/8ae401a968465ddb5b120c1770da146d429b4ad6/frontend/src/metabase-lib/queries/drills/underlying-records-drill.js#L29

Repro test https://github.com/metabase/metabase/blob/67a136c6f45fe3526ef7b1801ffcb25a6407867e/frontend/src/metabase-lib/tests/drills-underlying-records.unit.spec.ts#L51

### PR DESCRIPTION
**Pull request by camsaul**
Fixes #36143

I verified this fix with the JS unit test as well.

### CODE REVIEW

**Review by metamben — APPROVED** [review]
LGTM

File: src/metabase/lib/drill_thru/underlying_records.cljc [hunk 1]
```diff
@@ -40,7 +40,10 @@
      :type       :drill-thru/underlying-records
      ;; TODO: This is a bit confused for non-COUNT aggregations. Perhaps it should just always be 10 or something?
      ;; Note that some languages have different plurals for exactly 2, or for 1, 2-5, and 6+.
-     :row-count  (if (number? value) value 2)
+     :row-count  (if (and (number? value)
+                          (not (neg? value)))
+                   value
+                   2)
```
[the remark this answers is not in the document]
Comment (camsaul) [reply src/metabase/lib/drill_thru/underlying_records.cljc:46]: if you look at the issue, the behavior requested is specifically that if it's `< 0`, we return `2`

[the remark this answers is not in the document]
Comment (metamben) [reply src/metabase/lib/drill_thru/underlying_records.cljc:56]: I suspect the test is overly specific. 🤷‍♂️

**Comment by camsaul** [comment]
Redshift test failures are unrelated, see https://metaboat.slack.com/archives/C5XHN8GLW/p1701772468810829. Getting this merged in to keep the train rolling

**Pull request merged**